nextcloudappstore/core/api/v1/release/parser.py

```diff
@@ -0,0 +1,122 @@
+import re
+import tarfile  # type: ignore
+import lxml.etree  # type: ignore
+from typing import Dict, Any
+
+from nextcloudappstore.core.api.v1.release import ReleaseConfig
+from rest_framework.exceptions import APIException  # type: ignore
+
+
+class MaxSizeAppMetadataXmlException(APIException):
+    pass
+
+
+class InvalidAppMetadataXmlException(APIException):
+    pass
+
+
+class UnsupportedAppArchiveException(APIException):
+    pass
+
+
+class InvalidAppPackageStructureException(APIException):
+    pass
+
+
+class GunZipAppMetadataExtractor:
+    def __init__(self, config: ReleaseConfig) -> None:
+        """
+        :argument config the config
+        """
+        self.config = config
+        self.app_folder_regex = re.compile(r'^[a-z]+[a-z_]*$')
```

Comment:
ownCloud can only deal with these characters for apps. PHP autoloader suckage :)